### DISCUSSION TITLE
[Support] Cloudflare Proxies and DDNS w/PiVPN

### DISCUSSION THREAD

**Discussion by W00glin** [pivpn/pivpn#1494] [Support] [answered]
## In raising this issue, I confirm the following:

`{please fill the checkboxes, e.g: [X]}`

- [X] I have read the [documentation](https://docs.pivpn.io)
- [X] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [] The issue I am reporting can be *replicated*.
- [] The issue I am reporting can be *is* directly related to the pivpn installer script.

### Have you searched for similar issues and solutions?
    (yes/no / which issues?)
Yes. However none of the issues that are similar seemed close enough for reference.
### Has your install failed? 
```
No
```

#### What type of hardware are you running PiVPN at?

```
Raspberry Pi 3 Model B Plus Rev 1.3
```
#### Output of `uname -a`
```
Linux raspberrypi 4.19.66-v7+ #1253 SMP Thu Aug 15 11:49:46 BST 2019 armv7l GNU/Linux

```

#### Output of `cat /etc/os-release`
```
PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```
### If install failed Please provide the console output of  `curl -L https://install.pivpn.io | bash`
```
Did not fail.
```

### Issue

I am attempting to install and configure PiVPN on my raspberry pi utilizing cloudflares DDNS. I finally got tired of free DDNS providers and purchased my own domain. For this sake we shall call it `my-domain.net`. I followed the [NetworkChuck video](https://www.youtube.com/watch?v=rI-XxnyWFnM) for setting up DDNS on my raspberry pi via a bash script and a cronjob. 

Here is the kick. The PiVPN works, when I do not have the cloudflare proxies enabled. However, I would obviously like to have my IP address proxied. Is that an option? I'm not a networking guru and am still trying to wrap my head around DNS records. But my current cloudflare DNS dashboard looks like so:

| A           | my-domain.net            |  X.X.X.X (Real IP redacted)  | Proxied | Auto |
| CNAME | vpn  |  my-domain.net  | Proxied                             | Auto               |
| CNAME | www |  my-domain.net | Proxied                              | Auto              | 

Disabling proxies on all of the records and then stopping and starting the WireGuard connection on my phone allows it work. But, am I able to use the proxy services Cloudflare provides?

### Console output of      `curl -L install.pivpn.io | bash`
```
N/A
```

### Console output of      `pivpn add` or `pivpn add nopass`
```
N/A
```

### Console output of      `pivpn debug`

```
  ::: Generating Debug Output
::::            PiVPN debug              ::::
=============================================
::::            Latest commit            ::::
Branch: master
Commit: f80b0a7962d91862132c0a4abd65c1e67bd37bd7
Author: 4s3ti
Date: Fri Dec 3 21:41:13 2021 +0100
Summary: Update Changelog
=============================================
::::        Installation settings        ::::
PLAT=Raspbian
OSCN=stretch
USING_UFW=0
IPv4dev=eth0
IPv4addr=192.168.1.125/24
IPv4gw=192.168.1.1
install_user=pi
install_home=/home/pi
VPN=wireguard
pivpnPORT=2299
pivpnDNS1=1.1.1.1
pivpnDNS2=1.0.0.1
pivpnHOST=REDACTED
INPUT_CHAIN_EDITED=1
FORWARD_CHAIN_EDITED=1
pivpnPROTO=udp
pivpnMTU=1420
pivpnDEV=wg0
pivpnNET=10.6.0.0
subnetClass=24
ALLOWED_IPS="0.0.0.0/0, ::0/0"
UNATTUPG=1
INSTALLED_PACKAGES=()
=============================================
::::  Server configuration shown below   ::::
[Interface]
PrivateKey = server_priv
Address = 10.6.0.1/24
MTU = 1420
ListenPort = 2299
### begin Mobile ###
[Peer]
PublicKey = Mobile_pub
PresharedKey = Mobile_psk
AllowedIPs = 10.6.0.2/32
### end Mobile ###
=============================================
::::  Client configuration shown below   ::::
[Interface]
PrivateKey = Mobile_priv
Address = 10.6.0.2/24
DNS = 1.1.1.1, 1.0.0.1

```
### Have you taken any steps towards solving your issue?
```
  which?
```
Yes. I have tried googling and looking around and I have seen numerous blog post or guides that suggest that Pi-VPN with the Cloudflare proxies work but I have not found any specific guides outright stating it. Any ideas?

**Comment by coolapso** [comment]
Moving to discussions

**Reply by W00glin** [reply]
Wasn't sure if this fell under discussions or troubleshooting. Thanks!

**Reply by coolapso** [reply]
No worries. 
About the topic itself ... I personally can't really help with that ... hang on tight  maybe someone around knows something about it.

**Reply by W00glin** [reply]
@4s3ti no worries. Would it be more beneficial for the discussion if I edit the OP to just state what my problem is? Might help cut through the noise.

**Reply by coolapso** [reply]
yeah in this case should be fine. however i am not sure if the proxies really work for this scenario ... have you googled for the VPN protocol instead of PiVPN? might give you more results. 

PiVPN itself is not a VPN  ... it just installs a VPN protocol for you ... however i don't see how it can work with cloud-flare proxy ... but also i don't use any cloud-flare service anyways .. so i can't really tell.

**Reply by W00glin** [reply]
Right! I know its not a VPN unto itself, just a method of configuring VPNs. I just wanted to see if someone running PiVPN might have had a similar issue.

**Comment by TinCanTech** [comment]
`pvpn` is a script which installs **VPN applications**.

If those VPN applications work with your infra-structure is nothing to-do with `pivpn`.

The point is .. "googling and looking around" ..

Did you try asking cloudflare ?

**Reply by W00glin** [reply]
I am aware that `PiVPN `just installs and configures VPNs on the raspberry pi. What I am wondering is if anyone has had success using `PiVPNs `wireguard VPN config using while utilizing DDNS with cloudflare proxies.

The closest thing I found I found was this [guide](https://chrisjquinn.github.io/2021/10/PiVPN-with-DDNS/) and while I also tried adding a subdomain on Cloudflare to `vpn.my-domain.net` and then configuring the DNS name in the `pivpn` usingthe new `vpn.my-domain.net` that also, still did not work.

**Comment by EagleIJoe** [comment, marked as answer]
Sadly, this is not possible. The reply [here](https://community.cloudflare.com/t/wireguard-port-not-working/73492/2) explains it. Hence a separate DNS record in DNS mode (grey icon) for your VPN makes sense.

**Comment by Brennii96** [comment]
It might be possible using cloudflares zero trust dashboard where you can forward public domains to private networks without having to mess around with port fowarding or ddns although it doesn't look like they support UDP so won't work with Wireguard

**Reply by W00glin** [reply]
I had looked into this. But I just wanted a KISS solution so I opted not to have it proxied. Not ideal but I am not that stressed about it anyways. Thanks for the help.
